core-cube/src/main/java/org/apache/kylin/cube/CubeInstance.java

```diff
@@ -717,4 +724,51 @@ public static CubeSegment findSegmentWithJobId(String jobID, CubeInstance cubeIn
         throw new IllegalStateException("No segment's last build job ID equals " + jobID);
     }
 
+    public CubeInstance findLastSnapshot(List<RealizationEntry> realizationEntries, String lookupTableName) {
```

Comment:
CubeInstance is not the best place to put such a method. Please move it to CubeManager.